bluesky_widgets/models/run_engine_client.py

```diff
@@ -110,6 +155,296 @@ def load_re_manager_status(self, *, enforce=False):
+            #   Selection will be cleared when the table is reloaded, so save it in local variable
+            selected_uid = self.selected_queue_item_uid
+            if self.queue_item_uid_to_pos(selected_uid) < 0:
+                selected_uid = ""
+
+            # Update the representation of the queue
+            self.events.plan_queue_changed(
+                plan_queue_items=self._plan_queue_items,
+                running_item=self._running_item,
+            )
+
+            # Update selected item uid and emit 'queue_item_selection_changed' event
+            self._selected_queue_item_uid = selected_uid
+            self.events.queue_item_selection_changed(selected_item_uid=selected_uid)
+
+        except Exception as ex:
+            print(f"Exception: {ex}")
+
+    def load_plan_history(self):
+        try:
+            result = self._client.send_message(
+                method="history_get", raise_exceptions=True
+            )
+            if result["success"] is False:
+                raise RuntimeError(f"Failed to load history: {result['msg']}")
+            self._plan_history_items.clear()
+            self._plan_history_items.extend(result["items"])
+            self._plan_history_uid = result["plan_history_uid"]
+
+            # Deselect queue history if it does not exist in the queue
+            #   Selection will be cleared when the table is reloaded, so save it in local variable
+            selected_item_pos = self.selected_history_item_pos
+            if selected_item_pos >= len(self._plan_history_items):
+                selected_item_pos = -1
+
+            self.events.plan_history_changed(
+                plan_history_items=self._plan_history_items,
+            )
+
+            # Update selected item uid and emit 'queue_item_selection_changed' event
+            self._selected_history_item_pos = selected_item_pos
+            self.events.history_item_selection_changed(
+                selected_item_pos=selected_item_pos
+            )
+
+        except Exception as ex:
+            print(f"Exception: {ex}")
+
+    # ============================================================================
+    #                         Queue operations
+
+    @property
+    def selected_queue_item_uid(self):
+        return self._selected_queue_item_uid
+
+    @selected_queue_item_uid.setter
+    def selected_queue_item_uid(self, item_uid):
+        if self._selected_queue_item_uid != item_uid:
+            self._selected_queue_item_uid = item_uid
+            self.events.queue_item_selection_changed(selected_item_uid=item_uid)
+
+    def queue_item_uid_to_pos(self, item_uid):
+        # Returns -1 if item was not found
+        return self._plan_queue_items_pos.get(item_uid, -1)
+
+    def queue_item_pos_to_uid(self, n_item):
+        try:
+            item_uid = self._plan_queue_items[n_item]["item_uid"]
+        except Exception:
+            item_uid = ""
```

Comment:
This could catch unrelated errors and make them challenging to debug.
```suggestion
        try:
            item = self._plan_queue_items[n_item]
        except KeyError:
            item_uid = ""
        else:
            item_uid = item["item_uid"]
```

Reply:
My idea was to ignore `item_uid` if for some reason a corrupt item is received from the server. It doesn't happen in practice, but the application should not crash if it happens. The item can still be displayed. But I guess the proper error message should be displayed, so two conditions should be checked separately. I didn't work on error reporting yet.